src/main/java/com/raisetech/mybatisdemo/service/UserServiceImpl.java

```diff
@@ -38,11 +38,19 @@ public User createUser(String name, String address, Integer age) {
 
     @Override
     public void updateUser(User updateUser) {
-        userMapper.updateUser(updateUser);
+        User user = userMapper.findById(updateUser.getId())
+                .orElseThrow(() -> new ResourceNotFoundException("resource not found"));
+        if (Objects.nonNull(user)) {
+            userMapper.updateUser(updateUser);
+        }
```

Comment:
直前でorElseThrowしており、userがnullになることはないのでこの分岐は不要かと思います。
```suggestion
        userMapper.updateUser(updateUser);
```

Reply:
> 直前でorElseThrowしており、userがnullになることはないのでこの分岐は不要かと思います。

userMapper.findById(updateUser.getId())の結果に値が有る場合はorElseThrowでそのまま値が返されると思いますが、値がなかった場合は「ResourceNotFoundException("resource not found")」がuserに格納されるということでしょうか？

Reply:
> 値がなかった場合は「ResourceNotFoundException("resource not found")」がuserに格納されるということでしょうか？

ここの認識に相違がありますね。
例外がthrowされるときの挙動としては処理が
https://github.com/kima-s/lecture09/blob/cff6e33ed49141678fc8277933bc0732a5b7cf70/src/main/java/com/raisetech/mybatisdemo/service/UserServiceImpl.java#L41
ここで中断されて呼び出し元に処理が戻されます。

ただ、呼び出し元のControllerで例外をcatchしていないので更に呼び出し元にthrowされ最終的には

https://github.com/kima-s/lecture09/blob/cff6e33ed49141678fc8277933bc0732a5b7cf70/src/main/java/com/raisetech/mybatisdemo/controller/UserController.java#L65
こちらの例外ハンドラーで例外発生時の処理がなされます。

デバッガーで確認いただくとわかるのですが、

https://github.com/kima-s/lecture09/blob/cff6e33ed49141678fc8277933bc0732a5b7cf70/src/main/java/com/raisetech/mybatisdemo/service/UserServiceImpl.java#L41

にブレークポイントをあてて、存在しないIDでPATCHのリクエストを送るとここで処理が止まるはずです。
その後ステップオーバーすると次の処理には進まないはずです。

※Spring Bootのアプリケーションでのデバッグ方法が分からなければ聞いてください。
チャットでもいいですし、授業にそこを聞くためだけに来てくれても大丈夫です。

Reply:
> ここで中断されて呼び出し元に処理が戻されます。

ここがわかっていませんでした💦例外処理を実装している状態でエラーが発生すると、処理が中断されて、呼び出し元に例外がthrowされるのですね！

デバッグ行ってみて、確かに次の処理には進みませんでした！
ただ、Spring Bootのアプリケーションでデバッグを行うとファイルが複雑な為か、見たことのないファイルに次々と移っていくので、自分が確認したいことが確認できないもどかしさを感じています。もっと自分の確認したいところだけ確認する方法が有るのでしょうか？

Reply:
> ただ、Spring Bootのアプリケーションでデバッグを行うとファイルが複雑な為か、見たことのないファイルに次々と移っていくので、

Spring内部の実装に処理が移っていくので読んでもなにがなにやらとなりますよね。

> もっと自分の確認したいところだけ確認する方法が有るのでしょうか？

具体的にどこを確認したいのでしょうか？

Reply:
> Spring内部の実装に処理が移っていくので読んでもなにがなにやらとなりますよね。

本当にそう思います！！
例外が発生した際にどういう流れで@ExceptionHandlerにthrowされているのか等をデバッグで確認したかったりしたのですが、どうしてもspring内部の実装に入ると訳が分からなくなってしまい流れを掴めないでいました。良い方法でもあればと思ったのですが・・